### PR TITLE
Prioritise primary buildings in CycleBasesHotkeyLogic.

### LINKED ISSUE #20552: D2k: Setting primary Construction Yard doesn't work.
Setting your ConYard that isn't your first one as your primary one doesn't work as the "Jump to base" hotkey jumps to the first one anyway, except for the "Current facility" hotkey (but that hotkey applies to other types of facilities.)

### PR DESCRIPTION
**Pull request by pchote**
Fixes #20552.

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/Ingame/Hotkeys/CycleBasesHotkeyLogic.cs [hunk 1]
```diff
@@ -41,6 +41,8 @@ protected override bool OnHotkeyActivated(KeyInput e)
 
 			var bases = world.ActorsHavingTrait<BaseBuilding>()
 				.Where(a => a.Owner == player)
+				.OrderByDescending(a => a.IsPrimaryBuilding())
+				.ThenByDescending(a => a.ActorID)
```
Comment (dragunoff) [inline OpenRA.Mods.Common/Widgets/Logic/Ingame/Hotkeys/CycleBasesHotkeyLogic.cs]: This changes the behavior we have now (cycling from oldest to newest conyard). I think it makes more sense to cycle from oldest to newest.

**Review by dragunoff — CHANGES_REQUESTED** [review]

File: OpenRA.Mods.Common/Widgets/Logic/Ingame/Hotkeys/CycleBasesHotkeyLogic.cs (hunk 1 above)
Comment (PunkPun) [reply OpenRA.Mods.Common/Widgets/Logic/Ingame/Hotkeys/CycleBasesHotkeyLogic.cs]: does it? ActorID should reflect the age of the actor

Comment (pchote) [reply OpenRA.Mods.Common/Widgets/Logic/Ingame/Hotkeys/CycleBasesHotkeyLogic.cs]: Fixed.

**Review by PunkPun — APPROVED** [review]
lgtm

**Review by dragunoff — APPROVED** [review]

**Comment by PunkPun** [comment]
[Changelo](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/222afde629b5e42878e5b65bb4203deb2a3e88f3...71e3fd5f5faf86d6b86c921d293ebb50b9db8946)

**Pull request merged**